app/data_model/answer_store.py

```diff
@@ -264,6 +265,9 @@ def remove_by_location(self, location):
 
         self.remove(group_id=location.group_id, block_id=location.block_id, group_instance=location.group_instance)
 
+    def get_hash(self):
+        return hash(json.dumps(self.answers))
```

Comment:
Not sure if this is really something to be concerned with here, but I don't believe that json.dumps is guaranteed to return the same JSON string for the same Python dict each time (e.g. key orders etc.) even if logically the dict is identical. Relying on this to check if something has changed could lead to false-positives where the same dict will look like it has changed because the hash of the raw string will be different even if the actual data isn't. 

Again, we may not care?

Reply:
@ajmaddaford @jonnyshaw89 @jcox-dev  thoughts?

Reply:
Does it work?

Reply:
It does but we could ensure it's accurate each time with the `sort_keys` flag if we think that's necessary:

```python
    def get_hash(self):
        return hash(json.dumps(self.answers, sort_keys=True))
```

Reply:
I'd query the performance hit of this kind of thing, how often will this code run?

Reply:
Shouldn't take more than about 0.1ms (0.15ms with key sorting) to calculate the hash.